File: fizz-buzz/fizz_buzz.rb

def fizz_buzz(line)
  arr = line.split(' ')
  x = arr[0].to_i
  y = arr[1].to_i
  n = arr[2].to_i
  res = []

  (1..n).each do |i|
    s = ''
    s << 'F' if i % x == 0
    s << 'B' if i % y == 0
    if s.empty?
      res << i
    else
      res << s
    end
  end

  return res.join(' ')
end

File.open(ARGV[0]).each_line do |line|
  puts fizz_buzz(line) unless line.empty?
endr

File: fizz-buzz/fizz_buzz.js

var fs  = require("fs");

fs.readFileSync(process.argv[2]).toString().split('\n').forEach(function (line) {
  if (line !== "") {
    console.log(fizzBuzz(line));
  }
});

function fizzBuzz(line) {
  var arr = line.split(' ')
    , x   = parseInt(arr[0])
    , y   = parseInt(arr[1])
    , n   = parseInt(arr[2])
    , res = [];

  for (var i = 1; i <= n; ++i) {
    var s = '';
    if (i % x === 0)  s += 'F';
    if (i % y === 0)  s += 'B';
    if (s.length > 0) res.push(s);
    else              res.push(i);
  }

  return res.join(' ');
}